src/linker.rs

```diff
@@ -30,16 +30,22 @@ trait LinkerInterface {
 impl Linker {
     pub fn new(target: &str, linker: Option<&str>) -> Result<Linker, LinkerError> {
         let target_os = target.split('-').collect::<Vec<&str>>()[2];
-        let linker: Box<dyn LinkerInterface> = if let Some(linker) = linker {
-            Box::new(CcLinker::new(linker))
-        } else {
-            match target_os {
-                "linux" | "gnu" => Ok(Box::new(LdLinker::new())),
-                // "win32" | "windows" => Ok(Box::new(CcLinker::new("clang".to_string()))),
-                _ => Err(LinkerError::Target(target_os.into())),
-            }?
-        };
-        Ok(Linker { errors: Vec::default(), linker })
+
+        Ok(Linker {
+            errors: Vec::default(),
+            linker: match linker {
+                Some(linker) => Box::new(CcLinker::new(linker)),
+
+                // TODO: We should probably replace this match statement with an if one, because we would
+                // otherwise need to match against every exact e.g. MacOS version
+                None => match target_os {
+                    "linux" | "gnu" => Box::new(LdLinker::new()),
+                    "darwin22.1.0" => Box::new(CcLinker::new("clang")),
```

Comment:
The LD linker will work fine here and for now should be the default. Also if there's a way to avoid the version number it would be better maybe we can invert this Statement and only say windows does not get a linker. The idea was that Windows could not link using LD because we couldn't get it to accept all the correct settings at the time.

Reply:
Not sure why I didn't try `LdLinker` first but you're right, it's working :)
Anyways I don't have a Windows machine right now but can we assume `target_os` to always be `windows` or `win32`?

Reply:
windows and win32 are the only ones AFAIK